scripts/events_module/short/handle_short_events.py

```diff
@@ -399,8 +401,24 @@ def handle_accessories(self, pelts=Pelt):
                     if acc in acc_list:
                         acc_list.remove(acc)
 
-        if acc_list:
-            self.main_cat.pelt.accessory = random.choice(acc_list)
+        acc_categories = ["collars", "head_accessories", "tail_accessories", "body_accessories"]
+
+        if self.main_cat.pelt.accessory:
+            for acc in self.main_cat.pelt.accessory:
+                for category in acc_categories:
+                    if acc in getattr(pelts, category):
+                        for accessory in getattr(pelts, category):
+                            if accessory in acc_list:
+                                acc_list.remove(accessory)
+
```

Comment:
Horrifying.
(/silly)

It looks like this is trying to remove accessories that are in an already-occupied slot? Had to sit here and think about it for a hot minute buuut I think this is good? Please verify, I haven't run this code so it might explode!

```suggestion
accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]

for acc in self.main_cat.pelt.accessory:
    # find which accessory group it belongs to
    for i, lst in enumerate(accessory_groups):
        if acc in lst:
            # remove that group from possible accessories
            acc_list = [a for a in acc_list if a not in accessory_groups[i]]
            break
```

Reply:
Oh thank goodness you know of a better way of doing that, I knew it was ugly af but I couldn't figure out how to make it better. I shall test that change tomorrow!

Reply:
Tested and it works just fine! I just had to add a previous check to ensure the cat already has accessories